VisualizationBase/src/items/ViewItem.cpp

```diff
@@ -142,6 +130,63 @@ Model::Node* ViewItem::nodeAt(int column, int row) const
 	return nodes_[column][row];
 }
 
+void ViewItem::notifyAboutRemoval(Item *item)
+{
+	Q_ASSERT(item);
+	removeArrowsForItem(item);
+}
+
+void ViewItem::notifyAboutRemoval(Model::Node *node)
+{
+	Q_ASSERT(node);
+	//If somebody's spacing depends on the node or a child, remove
+	//the node's spacing target
+	QList<ViewItemNode*> spacingNodes;
+	for (auto node : allNodes())
+		if (auto viewNode = DCast<ViewItemNode>(node))
+			if (!viewNode->reference())
+				spacingNodes.append(viewNode);
+
+	QList<Model::Node*> toCheck{node};
+	while (!toCheck.isEmpty())
+	{
+		auto current = toCheck.takeFirst();
+		toCheck.append(current->children());
+		for (auto spacing : spacingNodes)
+			if (spacing->spacingParent() == current
+					|| spacing->spacingTarget() == current)
+			{
+				spacing->setSpacingTarget(nullptr);
+				spacing->setSpacingParent(nullptr);
+			}
+	}
+	//Remove all the info nodes depending on the node
+	QList<ViewItemNode*> infoNodeParents;
+	for (auto node : allNodes())
+	{
+		auto viewNode = DCast<ViewItemNode>(node);
+		if (DCast<InfoNode>(viewNode->reference()))
+			infoNodeParents.append(viewNode);
+	}
+
+	toCheck.append(node);
+	while (!toCheck.isEmpty())
+	{
+		auto current = toCheck.takeFirst();
+		toCheck.append(current->children());
```

Comment:
Hmm, why not just reuse the while loop above and handle both spacing and info nodes at once? This way you save yourself code duplication and redundant iteration.